api/v1beta1/types.go

```diff
@@ -737,10 +795,24 @@ func (s SubnetSpec) IsNatGatewayEnabled() bool {
 // virtual machine or virtual machine scale set.
 type SecurityProfile struct {
 	// This field indicates whether Host Encryption should be enabled
-	// or disabled for a virtual machine or virtual machine scale
-	// set. Default is disabled.
+	// or disabled for a virtual machine or virtual machine scale set.
+	// This should be disabled when SecurityEncryptionType is set to DiskWithVMGuestState.
+	// Default is disabled.
 	// +optional
 	EncryptionAtHost *bool `json:"encryptionAtHost,omitempty"`
+	// SecureBoot defines whether the instance should have secure boot enabled.
+	// Secure Boot verifies the digital signature of all boot components and halts the boot process if signature verification fails.
+	// If omitted, the platform chooses a default, which is subject to change over time, currently that default is Disabled.
+	// +kubebuilder:validation:Enum=Enabled;Disabled
+	//+optional
+	SecureBoot SecureBootPolicy `json:"secureBoot,omitempty"`
+	// VirtualizedTrustedPlatformModule enable virtualized trusted platform module measurements to create a known good boot integrity policy baseline.
+	// The integrity policy baseline is used for comparison with measurements from subsequent VM boots to determine if anything has changed.
+	// This is required to be set to Enabled if SecurityEncryptionType is defined.
+	// If omitted, the platform chooses a default, which is subject to change over time, currently that default is Disabled.
+	// +kubebuilder:validation:Enum=Enabled;Disabled
+	// +optional
+	VirtualizedTrustedPlatformModule VirtualizedTrustedPlatformModulePolicy `json:"virtualizedTrustedPlatformModule,omitempty"`
```

Comment:
I love this, thanks for such a detail explanation!
I am mostly in favor of Adhering to Azure REST API, but will also like to get opinion of @CecileRobertMichon. 

So it's a _yes_ from me to all of your questions
> 1. If there is no other objections, I will adjust the PR to the changes suggested. 
> 2. I will rename and convert these types to *bool

--- 

For 
> The Azure Go SDK defines this enum [here](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/resourcemanager/compute/armcompute/constants.go#L1685-L1698), so I would just add the same here, right?

You are right, a enum should work.

Reply:
+1 on adhering to Azure API. 

For the record this isn't the first time we run into this (Terraform being different), for example, we had the same dilemma with VM images. In this case I'm leaning toward the Azure API's explicit implementation because the name of the user-facing feature is `Confidential VM` so it is IMO slightly more user friendly to have that explicitly defined.

And yes we can use an enum for validation purposes. See https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/api/v1beta1/types.go#L262 for prior art.

Reply:
@nawazkh thank you for your feedback regarding the refactoring of adhering closely to the Azure REST API. Please take a look when possible, I believe the suggested refactoring is now in place.